eo-maven-plugin/src/test/java/org/eolang/maven/optimization/OptCachedTest.java

```diff
@@ -0,0 +1,113 @@
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven.optimization;
+
+import com.jcabi.xml.XML;
+import com.jcabi.xml.XMLDocument;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.cactoos.io.ResourceOf;
+import org.cactoos.io.UncheckedInput;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.hamcrest.io.FileMatchers;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+/**
+ * Test case for {@link org.eolang.maven.optimization.OptCached}.
+ * @since 0.28.12
+ */
+class OptCachedTest {
+    @Test
+    void optimizesIfXmlAlreadyInCache(final @TempDir Path tmp) {
+        final XML program = OptCachedTest.program();
+        MatcherAssert.assertThat(
+            new OptCached(path -> program, tmp).apply(OptCachedTest.save(tmp, program)),
+            Matchers.equalTo(program)
+        );
+    }
+
+    @Test
+    void optimizesIfXmlIsAbsentInCache(final @TempDir Path tmp) {
+        final XML program = OptCachedTest.program();
+        final Path cache = tmp.resolve("cache");
+        final XML res = new OptCached(path -> program, cache)
+            .apply(OptCachedTest.save(tmp, program));
+        MatcherAssert.assertThat(
+            res,
+            Matchers.equalTo(program)
+        );
+        MatcherAssert.assertThat(
+            cache.resolve("main.xmir").toFile(),
+            FileMatchers.anExistingFile()
+        );
+        MatcherAssert.assertThat(
+            res, Matchers.equalTo(program)
+        );
+    }
+
+    @Test
+    void throwsExceptionIfCantFindProgram(final @TempDir Path tmp) {
+        Assertions.assertThrows(
+            IllegalStateException.class,
+            () -> new OptCached(p -> OptCachedTest.program(), tmp.resolve("cache"))
+                .apply(tmp.resolve("non-existent"))
+        );
+    }
+
+    /**
+     * Save XML program with hardcoded name to a temp directory.
+     * @param tmp Temporary test directory.
+     * @param xml XML program.
+     * @return Path to saved program.
+     */
+    private static Path save(final Path tmp, final XML xml) {
+        final Path res = tmp.resolve("main.xmir");
+        try {
+            Files.write(res, xml.toString().getBytes(StandardCharsets.UTF_8));
```

Comment:
@volodya-lombrozo why can't you use our object `Home`, which knows how to save into file?

Reply:
@yegor256 done.